include/lbann/callbacks/checkpoint.hpp

```diff
@@ -186,17 +197,20 @@ static inline bool write_latest(std::string filename, int epoch, int train) {
 /** \brief Reads the "latest" file and returns the epoch number and
  *        sample offset for most recent checkpoint
  */
-static inline bool read_latest(std::string filename, int *epochLast, int *trainLast) {
+inline bool read_latest(std::string filename, execution_mode *mode, size_t *epochLast, size_t *trainLast) {
```

Comment:
This can be delayed to the C++-ification if it's easier, but these should all be pass-by-reference. 
```suggestion
inline bool read_latest(std::string filename, execution_mode& mode, size_t& epochLast, size_t& trainLast) {
```

Reply:
#1225